c2c: deflake and refactor TestTenantReplicationProtectedTimestampManagement

msbutler opened the pull request:
In #106473, the test harness attempted to cutover to a timestamp that was older than the replication job's retained timestamp. This doesn't seem that suprising since the ttl in the test is 1 second.

To prevent future flakes, the harness now cuts over to the latest replicated time.

Fixes #106473

Release note: None

msbutler committed:
c2c: deflake TestTenantReplicationProtectedTimestampManagement

In #106473, the test harness attempted to cutover to a timestamp that was half
a second older than the replication job's retained timestamp. This failure
isn't worrisome since the retention ttl of the ingestion job is 1 second.

To prevent future flakes, the harness now cuts over to the latest replicated
time.

Fixes #106473

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/106561)
<!-- Reviewable:end -->

msbutler committed:
c2c: refactor TestTenantReplicationProtectedTimestampManagement

This patch makes the test slightly more readible by refactoring the
testProtectedTimestampManagment into an anonymous function to be called by
testUtils.RunTrueAndFalse. It also shortens the test name to
TestProtectedTimestampManagement.

Release note: None

Epic: none

stevendanna commented on pkg/ccl/streamingccl/replicationtestutils/testutils.go:
```diff
@@ -171,16 +171,23 @@ func (c *TenantStreamingClusters) WaitUntilStartTimeReached(ingestionJobID jobsp
-		c.Args.DestTenantName, cutoverTime).Scan(&cutoverStr)
-	cutoverOutput := DecimalTimeToHLC(c.T, cutoverStr)
-	require.Equal(c.T, cutoverTime, cutoverOutput.GoTime())
+	if cutoverTime.Equal(time.Time{}) {
```
If you fancy: `cutoverTime.IsZero()`?

msbutler replied:
done.

stevendanna reviewed (approved):
Thanks!

msbutler commented:
in the update, I also shortened the te test name to TestProtectedTimestampManagement.

msbutler commented:
TFTR!

bors r=stevendanna
